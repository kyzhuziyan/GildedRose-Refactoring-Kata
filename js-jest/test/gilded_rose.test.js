const { Shop, Item } = require("../src/gilded_rose");

const getExamples = () => {
  const name_lst = [
    "foo",
    "Aged Brie",
    "Backstage passes to a TAFKAL80ETC concert",
    "Conjured",
  ];
  const sell_in_lst = [-1, 0, 1, 5, 10, 15, 2];
  const quality_lst = [0, 1, 5, 10, 15, 49, 50];

  const examples = name_lst
    .map((name) =>
      sell_in_lst.map((sell_in) =>
        quality_lst.map((quality) => {
          const itemArgs = new Item(name, sell_in, quality);
          return itemArgs;
        })
      )
    )
    .flat(2);
  return [
    ...examples,
    new Item("Sulfuras, Hand of Ragnaros", 0, 80),
    new Item("Sulfuras, Hand of Ragnaros", -1, 80),
    new Item("Sulfuras, Hand of Ragnaros", 10, 80),
    undefined,
  ];
};

describe("snapshot test", function () {
  const examples = getExamples();
  const output = [];
  examples.forEach((example) => {
    it("items", function () {
      const gildedRose = new Shop(example ? [example] : undefined);
      const items = gildedRose.updateQuality();
      output.push([...items]);
    });
  });
  expect(output).toMatchSnapshot();

  it("snapshot test", () => {
    expect(output).toMatchSnapshot();
  });
});
