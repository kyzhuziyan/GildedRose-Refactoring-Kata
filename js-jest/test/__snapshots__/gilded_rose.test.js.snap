// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `Array []`;

exports[`snapshot test snapshot test 1`] = `
Array [
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 3,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 8,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 13,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 47,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 48,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 3,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 8,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 13,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 47,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 48,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 4,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 9,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 14,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 48,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 49,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 4,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 9,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 14,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 48,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 49,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 4,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 9,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 14,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 48,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 49,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 4,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 9,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 14,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 48,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 49,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 0,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 4,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 9,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 14,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 48,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "foo",
      "quality": 49,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 3,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 7,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 12,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 17,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 3,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 7,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 12,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 17,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 1,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 6,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 11,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 16,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 1,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 6,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 11,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 16,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 1,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 6,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 11,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 16,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 1,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 6,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 11,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 16,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 1,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 2,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 6,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 11,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 16,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Aged Brie",
      "quality": 50,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 3,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 4,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 8,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 13,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 18,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 3,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 4,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 8,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 13,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 18,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 2,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 3,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 7,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 12,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 17,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 1,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 2,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 6,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 11,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 16,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 3,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 4,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 8,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 13,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 18,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Backstage passes to a TAFKAL80ETC concert",
      "quality": 50,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 3,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 8,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 13,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 47,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 48,
      "sellIn": -2,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 3,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 8,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 13,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 47,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 48,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 3,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 8,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 13,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 47,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 48,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 3,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 8,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 13,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 47,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 48,
      "sellIn": 4,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 3,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 8,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 13,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 47,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 48,
      "sellIn": 9,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 3,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 8,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 13,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 47,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 48,
      "sellIn": 14,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 0,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 3,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 8,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 13,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 47,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Conjured",
      "quality": 48,
      "sellIn": 1,
    },
  ],
  Array [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 0,
    },
  ],
  Array [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": -1,
    },
  ],
  Array [
    Item {
      "name": "Sulfuras, Hand of Ragnaros",
      "quality": 80,
      "sellIn": 10,
    },
  ],
  Array [],
]
`;
